equinix/resource_metal_device.go

```diff
@@ -270,18 +270,18 @@ func resourceMetalDevice() *schema.Resource {
 
 			"user_data": {
 				Type:        schema.TypeString,
-				Description: "A string of the desired User Data for the device",
+				Description: "A string of the desired User Data for the device.  By default, changing this attribute will cause terraform to destroy and recreate your device.  If `reinstall` is specified or `behavior.allow_changes` includes `\"user_data\"`, the device will be updated in-place instead of recreated.",
```

Comment:
This documentation is reused in Terraform based providers. I think we'll get more mileage in referring to the code/project as "the provider". 

> changing this attribute will cause _the provider_ to destroy ...